mne/io/nsx/nsx.py

```diff
@@ -0,0 +1,539 @@
+
+        # if the NSx info contained channel type information
+        # set it now
+        ch_type = ch_types[idx]
+        if ch_type is not None and ch_type in CH_TYPE_MAPPING:
+            chan_info['kind'] = CH_TYPE_MAPPING.get(ch_type)
+            if ch_type not in ['EEG', 'ECOG', 'SEEG', 'DBS', 'CC']:
+                chan_info['coil_type'] = FIFF.FIFFV_COIL_NONE
+                pick_mask[idx] = False
+        # if user passes in explicit mapping for eog, misc and stim
+        # channels set them here
+        if ch_name in eog or idx in eog or idx - nchan in eog:
+            chan_info['coil_type'] = FIFF.FIFFV_COIL_NONE
+            chan_info['kind'] = FIFF.FIFFV_EOG_CH
+            pick_mask[idx] = False
+        elif ch_name in misc or idx in misc or idx - nchan in misc:
+            chan_info['coil_type'] = FIFF.FIFFV_COIL_NONE
+            chan_info['kind'] = FIFF.FIFFV_MISC_CH
+            pick_mask[idx] = False
+        elif idx in stim_channel_idxs:
+            chan_info['coil_type'] = FIFF.FIFFV_COIL_NONE
+            chan_info['unit'] = FIFF.FIFF_UNIT_NONE
+            chan_info['kind'] = FIFF.FIFFV_STIM_CH
+            pick_mask[idx] = False
+            chan_info['ch_name'] = ch_name
+            ch_names[idx] = chan_info['ch_name']
+        elif ch_type not in CH_TYPE_MAPPING:
+            chs_without_types.append(ch_name)
+        chs.append(chan_info)
+
+    # warn if channel type was not inferable
+    if len(chs_without_types):
+        msg = ('Could not determine channel type of the following channels, '
+               f'they will be set as EEG:\n{", ".join(chs_without_types)}')
+        logger.info(msg)
+
+    sfreq = nsx_info['timestamp_resolution'] / nsx_info['period']
+    info = _empty_info(sfreq)
+    info['meas_date'] = nsx_info['meas_date']
+    info['chs'] = chs
+    info['ch_names'] = ch_names
+
+    highpass = nsx_info['highpass'][:128]
+    lowpass = nsx_info['lowpass'][:128]
+    if np.all(highpass == highpass[0]):
+        if highpass[0] == 'NaN':
+            # Placeholder for future use. Highpass set in _empty_info.
+            pass
+        else:
+            hp = highpass[0]
+            try:
+                hp = float(hp)
+            except Exception:
+                hp = 0.
+            info['highpass'] = hp
+    else:
+        info['highpass'] = float(np.max(highpass))
+        warn('Channels contain different highpass filters. Highest filter '
+             'setting will be stored.')
+
+    if np.all(lowpass == lowpass[0]):
+        if lowpass[0] in ('NaN', '0', '0.0'):
+            # Placeholder for future use. Lowpass set in _empty_info.
+            pass
+        else:
+            info['lowpass'] = float(lowpass[0])
+    else:
+        info['lowpass'] = float(np.min(lowpass))
+        warn('Channels contain different lowpass filters. Lowest filter '
+             'setting will be stored.')
+    if np.isnan(info['lowpass']):
+        info['lowpass'] = info['sfreq'] / 2.
```

Comment:
set to `None` when it's not known? This doesn't seem wrong technically but I wonder if it's robust to resampling of the data

Reply:
Actually,`_empty_info()` already sets `info["lowpass"] = info["sfreq"] / 2.0`, so I am gonna remove this part.